bigquery-storage/api/BigQueryStorage.Samples/AppendRowsPending.cs

```diff
@@ -0,0 +1,154 @@
+        };
+
+        // Initialize the append row request
+        // Set an offset to allow resuming this stream if the connection breaks.
+        // In Production code, you may want to keep track of which requests the server has acknowledged
+        // and resume the stream at the first non-acknowledged message. If the server has already
+        // processed a message with that offset, it will return an ALREADY_EXISTS
+        // error, which can be safely ignored.
+        // The first request must always have an offset of 0.
+        AppendRowsRequest appendRowRequest = new AppendRowsRequest
+        {
+            WriteStreamAsWriteStreamName = stream.WriteStreamName,
+            Offset = 0L,
+            ProtoRows = protoData
+        };
+
+        // Stream a request to the server
+        await rowAppender.WriteAsync(appendRowRequest);
+
+        protoData = new ProtoData
+        {
+            Rows = new ProtoRows { SerializedRows = { new CustomerRecord { CustomerNumber = 3, CustomerName = "Charles" }.ToByteString() } }
+        };
+
+        // Since this is the second request, you only need to include the row data.
+        // The name of the stream and protocol buffers DESCRIPTOR is only needed in
+        // the first request.
+        appendRowRequest = new AppendRowsRequest
+        {
+            // Offset must equal the number of rows that were previously sent.
+            // If Offset is not present, the write is performed at the current end of stream. 
+            Offset = 2L,
+            ProtoRows = protoData
+        };
+
+        await rowAppender.WriteAsync(appendRowRequest);
+
+        // Complete writing requests to the stream
+        await rowAppender.WriteCompleteAsync();
+
+        // Await the handler. This will complete once all server responses have been processed
+        await appendResultsHandlerTask;
+
+        // A PENDING type stream must be "finalized" before being committed. No new
+        // records can be written to the stream after this method has been called.
+        await bigQueryWriteClient.FinalizeWriteStreamAsync(stream.Name);
+        BatchCommitWriteStreamsRequest batchCommitWriteStreamsRequest = new BatchCommitWriteStreamsRequest
+        {
+            Parent = table.ToString(),
+            WriteStreams = { stream.Name }
+        };
+
+        BatchCommitWriteStreamsResponse batchCommitWriteStreamsResponse =
+            await bigQueryWriteClient.BatchCommitWriteStreamsAsync(batchCommitWriteStreamsRequest);
+        // Commit Time is set only if batch commit is successful
+        if (batchCommitWriteStreamsResponse.CommitTime != null)
+        {
+            foreach (StorageError error in batchCommitWriteStreamsResponse.StreamErrors)
+            {
+                Console.WriteLine($"{error.ErrorMessage}");
+            }
+
+            // In Production, you may want to throw an exception stating error committing the streams
```

Comment:
Or retry the operation? (It's probably worth talking to the BigQuery team about what they want to advise here.)

Reply:
Sure. Let me find a point of contact and reach out to them. I will update as they advise.